sdk/tables/azure-data-tables/src/test/java/com/azure/data/tables/TableClientTest.java

```diff
@@ -0,0 +1,1044 @@
+import com.azure.core.http.rest.Response;
+import com.azure.core.test.TestBase;
+import com.azure.core.test.utils.TestResourceNamer;
+import com.azure.data.tables.models.ListEntitiesOptions;
+import com.azure.data.tables.models.TableAccessPolicies;
+import com.azure.data.tables.models.TableAccessPolicy;
+import com.azure.data.tables.models.TableEntity;
+import com.azure.data.tables.models.TableEntityUpdateMode;
+import com.azure.data.tables.models.TableSignedIdentifier;
+import com.azure.data.tables.models.TableTransactionAction;
+import com.azure.data.tables.models.TableTransactionActionResponse;
+import com.azure.data.tables.models.TableTransactionActionType;
+import com.azure.data.tables.models.TableTransactionFailedException;
+import com.azure.data.tables.models.TableTransactionResult;
+import com.azure.data.tables.sas.TableSasIpRange;
+import com.azure.data.tables.sas.TableSasPermission;
+import com.azure.data.tables.sas.TableSasProtocol;
+import com.azure.data.tables.sas.TableSasSignatureValues;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+
+import java.time.Duration;
+import java.time.OffsetDateTime;
+import java.time.ZoneOffset;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Date;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.fail;
+
+/**
+ * Tests {@link TableClient}.
+ */
+public class TableClientTest extends TestBase {
+    private TableClient tableClient;
+    private HttpPipelinePolicy recordPolicy;
+    private HttpClient playbackClient;
+
+    @Override
+    protected void beforeTest() {
+        final String tableName = testResourceNamer.randomName("tableName", 20);
+        final String connectionString = TestUtils.getConnectionString(interceptorManager.isPlaybackMode());
+
+        final TableClientBuilder builder = new TableClientBuilder()
+            .connectionString(connectionString)
+            .httpLogOptions(new HttpLogOptions().setLogLevel(HttpLogDetailLevel.BODY_AND_HEADERS))
+            .tableName(tableName);
+
+        if (interceptorManager.isPlaybackMode()) {
+            playbackClient = interceptorManager.getPlaybackClient();
+
+            builder.httpClient(playbackClient);
+        } else {
+            builder.httpClient(HttpClient.createDefault());
+            if (!interceptorManager.isLiveMode()) {
+                recordPolicy = interceptorManager.getRecordPolicy();
+
+                builder.addPolicy(recordPolicy);
+            }
+
+            builder.addPolicy(new RetryPolicy(new ExponentialBackoff(6, Duration.ofMillis(1500),
```

Comment:
Any reason for this custom retry policy?

Reply:
Not really, I can look into it a bit.

Reply:
I think it's there just to be super safe in live tests, but I don't thinks it's necessarily required. I'll get rid of it for now.